disnake/ext/commands/params.py

```diff
@@ -285,97 +365,87 @@ def _parse_enum(self, annotation: Any) -> None:
-                self._parse_enum(conv_annot)
-                return
-            elif conv_annot is not Any:
-                raise TypeError("Converters cannot use converter annotations")
-            elif annotation in CONVERTER_MAPPING:
-                raise TypeError(
-                    "Cannot use an implicit converter annotation and an unnanotated converter at the same time"
-                )
-            # otherwise just parse the annotation normally and hope for the best
+        if not converter_mode:
+            self.converter = (
+                self.converter
+                or getattr(annotation, "__discord_converter__", None)
+                or self._registered_converters.get(annotation)
+            )
+            if self.converter:
+                self.parse_converter_annotation(self.converter, annotation)
+                return True
 
         if annotation is inspect.Parameter.empty or annotation is Any:
-            pass
-        elif get_origin(annotation) is list:
-            if self.converter:
-                raise TypeError("Converter detected with custom annotation")
-            arg = annotation.__args__[0] if annotation.__args__ else str
-            if arg in [str, int, float]:
-                conv = arg
-            elif arg in CONVERTER_MAPPING:
-                # TODO: Define our own converters?
-                raise TypeError(
-                    "Discord's api is not mature enough to handle member conversion with models"
-                )
-            else:
-                raise TypeError(f"{arg!r} is not a valid List subscript for Param")
-            self.converter = lambda inter, arg: list(map(conv, arg.split()))
+            return False
+        elif annotation in self.TYPES:
+            self.type = annotation
         elif (
             isinstance(annotation, (EnumMeta, disnake.enums.EnumMeta))
             or get_origin(annotation) is Literal
         ):
             self._parse_enum(annotation)
-
         elif get_origin(annotation) is Union:
             args = annotation.__args__
-            if all(issubclass(channel, disnake.abc.GuildChannel) for channel in args):
-                self.type = disnake.abc.GuildChannel
-                if not self.channel_types:
-                    channel_types = set()
-                    for channel in args:
-                        channel_types.union(_channel_type_factory(channel))
-                    self.channel_types = list(channel_types)
-            elif annotation in self.TYPES:
-                self.type = annotation
-            elif any(get_origin(arg) for arg in args):
-                raise TypeError("Unions do not support nesting")
+            if all(issubclass_(channel, disnake.abc.GuildChannel) for channel in args):
+                self._parse_guild_channel(*args)
             else:
                 raise TypeError(
                     "Unions for anything else other than channels or a mentionable are not supported"
```

Comment:
Since unions are hashable and do not care about the order of their arguments this is handled by `Union[disnake.Member, disnake.Role]: 9` in `ParamInfo.TYPES`

Reply:
I see, makes sense